packages/flutter_tools/lib/src/resident_web_runner.dart

```diff
@@ -179,27 +221,80 @@ class ResidentWebRunner extends ResidentRunner {
       timeout: timeoutConfiguration.fastOperation,
       progressId: 'hot.restart',
     );
-    OperationResult result = OperationResult.ok;
-    try {
-      final List<Uri> invalidatedSources = ProjectFileInvalidator.findInvalidated(
-        lastCompiled: _lastCompiled,
-        urisToMonitor: <Uri>[
-          for (FileSystemEntity entity in flutterProject.directory
-              .childDirectory('lib')
-              .listSync(recursive: true))
-            if (entity is File && entity.path.endsWith('.dart')) entity.uri
-        ], // Add new class to track this for web.
-        packagesPath: PackageMap.globalPackagesPath,
-      );
-      await webCompilationProxy.invalidate(inputs: invalidatedSources);
-      await _connection.sendCommand('Page.reload');
-      await Future<void>.delayed(const Duration(milliseconds: 150));
-    } catch (err) {
-      result = OperationResult(1, err.toString());
-    } finally {
-      printStatus('Restarted application in ${getElapsedAsMilliseconds(timer.elapsed)}.');
-      status.cancel();
-    }
-    return result;
+    final Response response = await vmService.callMethod('hotRestart');
```

Comment:
Not clear: does the client talk to the build daemon directly, or do we need to separately send a build command?